src/open_inwoner/cms/plugins/views.py

```diff
@@ -0,0 +1,70 @@
+import itertools
+from typing import Sequence
+
+from django.http import HttpRequest, HttpResponse
+from django.shortcuts import render
+from django.views import View
+
+import structlog
+from ipware import get_client_ip
+
+from open_inwoner.cms.cases.views.mixins import CaseLogMixin
+from open_inwoner.cms.cases.views.services import CaseListService
+from open_inwoner.cms.plugins.models.zaken import MAX_CASES_DEFAULT, MIN_CASES
+from open_inwoner.htmx.mixins import RequiresHtmxMixin
+from open_inwoner.openzaak.types import UniformCase
+
+logger = structlog.stdlib.get_logger(__name__)
+
+
+class ZakenPluginContentView(RequiresHtmxMixin, CaseLogMixin, View):
+    """
+    HTMX endpoint that fetches and returns zaken content for the `CMSZakenPlugin`.
+
+    The view accepts and validates a `num_zaken` query parameter to limit the number
+    of cases displayed.
+    """
+
+    def get(self, request: HttpRequest, plugin_id: int) -> HttpResponse:
+        case_service = CaseListService(request)
+
+        open_submissions: Sequence[UniformCase] = case_service.get_submissions()
+        preprocessed_cases: Sequence[UniformCase] = case_service.get_cases()
+
+        try:
+            num_zaken = int(request.GET.get("num_zaken"))
+            if num_zaken < MIN_CASES or num_zaken > MAX_CASES_DEFAULT:
+                client_ip, _ = get_client_ip(request)
+                logger.warning(
+                    "Invalid num_zaken value, using default",
+                    num_zaken=num_zaken,
+                    client_ip=client_ip,
+                )
+                num_zaken = MAX_CASES_DEFAULT
+        except (ValueError, TypeError) as exc:
+            client_ip, _ = get_client_ip(request)
+            logger.warning(
+                "Invalid num_zaken parameter, using default",
+                value=request.GET.get("num_zaken"),
```

Comment:
```suggestion
                value=request.GET.get("num_zaken", None),
```

And maybe move this to the top too. We want to separate out the "absent" case from the "bad value" case.

Reply:
I think we need this change because it could still raise an exception if it's missing, or better yet:


```suggestion
                value=num_zaken,
```